### ISSUE TITLE
DeadLock on fitting

### ISSUE THREAD

**Issue by JasonBourne1998** [lijunxian111/DEMASQ-Pytorch-Version#2]
Hi, thanks for your implementation, but not sure why I got stuck on the   line 87, train.py: loss, preds = model.fit(x, y).

The tqdm seems not change:

![image](https://github.com/user-attachments/assets/3a39bae9-2f22-47fa-afdc-b217798ced2d)

**Comment by JasonBourne1998** [comment]
Seems not deadlock, it's a training slow problem here. May I know the training time ? Seems it will cost almost one hour to pass 1000 epoch.

**Comment by lijunxian111** [comment]
May I know what device you use? GPU or CPU?

**Comment by JasonBourne1998** [comment]
> May I know what device you use? GPU or CPU?

Hi， thanks for reply, I have a GPU device, but it seems that there is no code snippet like .to(device). So I think maybe it is run with cpu.

**Comment by JasonBourne1998** [comment]
Update: it seems that I already convert to GPU, but it is still time-comsuming task.
![image](https://github.com/user-attachments/assets/394fdc34-abbb-4042-a5ae-f36c02ca13a1)

**Comment by lijunxian111** [comment]
thanks for answering. I think，you can finish training for a few hours&nbsp;



---Original---
From: "LIAO ***@***.***&gt;
Date: Fri, Aug 9, 2024 14:56 PM
To: ***@***.***&gt;;
Cc: "Junxian ***@***.******@***.***&gt;;
Subject: Re: [lijunxian111/DEMASQ-Pytorch-Version] DeadLock on fitting (Issue#2)




  
May I know what device you use? GPU or CPU?
  
Hi， thanks for reply, I have a GPU device, but it seems that there is no code snippet like .to(device). So I think maybe it is run with cpu.
